react_ws_src/src/helpers/computer_moves.js

```diff
@@ -0,0 +1,168 @@
+const BOARD_SIZE = 9;
+const CELL_PREFIX = 'c';
+const PLAYER = 'x';
+const COMPUTER = 'o';
+
+/**
+ * Finds the best move for the computer using the minimax algorithm.
+ * It converts the move index into the cell id used within the UI.
+ *
+ * NOTE: ideally we wouldn't have to convert between the cell id and the index since that is a UI concern.
+ * Nevertheless, I felt it was better to encapsulate everything into 1 function. If I had more time I'd refactor
+ * the component a bit to create clearer separation of concerns between the move logic and the presentation.
+ */
+export function get_computer_move(current_moves) {
+  const board = convert_move_coordinates_to_move_position_array(normalize_move_coordinates(current_moves));
+  const best_move = get_minimax(board, COMPUTER);
+  return convert_index_to_cell(best_move.index);
+}
+
+/**
+ * Adds the cell prefix and convert to one based index
+ */
+export function convert_index_to_cell(index) {
+  return `${CELL_PREFIX}${index + 1}`;
+}
+
+
+/**
+ * Remove the cell prefix and convert to zero based index
+ */
+export function convert_cell_to_index(cell) {
+  return parseInt(cell.replace(CELL_PREFIX, ''), 10) - 1;
+}
+
+/**
+ * Converts the move coordinates object to a zero based array
+ */
+export function normalize_move_coordinates(current_moves) {
+  const normalized_moves = {};
+
+  for (let [cell, player] of Object.entries(current_moves)) {
+    normalized_moves[convert_cell_to_index(cell)] = player;
+  }
+
+  return normalized_moves;
+}
+
+/**
+ * Gets a ttt board of the given size
+ */
+export function get_board(size) {
+  return Array(size).fill().map((x, i) => i);
+}
+
+/**
+ * Given the current moves by each player applies them to the board array
+ */
+export function convert_move_coordinates_to_move_position_array(current_moves) {
+  const board = get_board(BOARD_SIZE);
```

Comment:
Even though I parametized the board size, I didn't put any other safe guards around it. ie. we should only allow odd numbers with perfect squares, otherwise the game becomes unplayable. I figured since the board size is hardcoded for now, it's not a big deal.